grub2: x86: efi boot add search module for better find boot partition on multiple disk

ShireCiel opened the pull request:
Previous `set root='(hd0,gpt1)'` on multiple disk may find a wrong disk.
Using grub search function and previous boot partition label to better find boot disk.
Label name "kernel" from scripts/gen_image_generic.sh.

ShireCiel commented:
Sign off and PR description already added.

wulfy23 commented:
modules are listed alpha-order... needs to shift back two

ShireCiel commented:
> modules are listed alpha-order... needs to shift back two

Not sure understand rules. Should I change title to "grub2: x86: efi boot add search module for better find boot partition on multiple disk"?
Is there have some example?
Thanks.

wulfy23 commented on package/boot/grub2/Makefile:
```diff
@@ -120,28 +120,28 @@ define Package/grub2/install
 		-c ./files/grub-early.cfg \
 		-o $(STAGING_DIR_IMAGE)/grub2/legacy-core.img \
-		biosdisk boot chain configfile ext2 linux ls part_msdos reboot serial vga
+		biosdisk boot chain configfile ext2 linux ls part_msdos reboot serial vga search
```
search is after reboot alphabetically

wulfy23 commented on package/boot/grub2/Makefile:
```diff
@@ -120,28 +120,28 @@ define Package/grub2/install
 		-c $(PKG_BUILD_DIR)/grub-early.cfg \
 		-o $(STAGING_DIR_IMAGE)/grub2/gpt-core.img \
-		at_keyboard biosdisk boot chain configfile fat linux ls part_gpt reboot serial vga
+		at_keyboard biosdisk boot chain configfile fat linux ls part_gpt reboot serial vga search
```
search is after reboot alphabetically

wulfy23 commented on package/boot/grub2/Makefile:
```diff
@@ -120,28 +120,28 @@ define Package/grub2/install
 		-c ./files/grub-early.cfg \
 		-o $(STAGING_DIR_IMAGE)/grub2/generic-core.img \
-		at_keyboard biosdisk boot chain configfile ext2 linux ls part_msdos reboot serial vga
+		at_keyboard biosdisk boot chain configfile ext2 linux ls part_msdos reboot serial vga search
```
search is after reboot alphabetically

wulfy23 commented on package/boot/grub2/Makefile:
```diff
@@ -120,28 +120,28 @@ define Package/grub2/install
 		-c ./files/grub-early.cfg \
 		-o $(STAGING_DIR_IMAGE)/grub2/eltorito.img \
-		at_keyboard biosdisk boot chain configfile iso9660 linux ls part_msdos reboot serial test vga
+		at_keyboard biosdisk boot chain configfile iso9660 linux ls part_msdos reboot serial test vga search
```
search is after reboot alphabetically

ShireCiel replied:
Understood!  Problem in codes. 😂. 
Thanks!

wulfy23 reviewed (commented):
hopefully clearer for you

ShireCiel committed:
grub: add search module for better find boot partition on multiple disk

Previous set root='(hd0,gpt1)' on multiple disk may find a wrong disk.
Using grub search function and previous boot partition label to better find boot disk.
Label name "kernel" from scripts/gen_image_generic.sh.

Signed-off-by: Jax Jiang <jax.jiang.007@gmail.com>

ShireCiel commented:
> hopefully clearer for you

thanks. edit complete.

bobafetthotmail commented:
Very nice. 

The MX-100 device in x86-64 target had this issue when booting efi so the contributor implemented a workaround
https://github.com/openwrt/openwrt/blob/master/target/linux/x86/base-files/lib/upgrade/platform.sh#L73
that changes 
set root='(hd0,gpt1)'
to
set root='(hd1,gpt1)'

This PR is fixing the problem and removing that line from the file, so the workaround can be removed. 
You can do an additional commit to do this change if you want.

bobafetthotmail reviewed (approved):
Very nice.

The MX100 device in x86-64 target had this issue when booting efi so the contributor implemented a workaround
https://github.com/openwrt/openwrt/blob/master/target/linux/x86/base-files/lib/upgrade/platform.sh#L73
that changes
set root='(hd0,gpt1)'
to
set root='(hd1,gpt1)'

This PR is fixing the problem and removing that line from the file, so the workaround for MX100 can be removed.
You can do an additional commit to do this change if you want.

mans0n commented:
Please wrap the commit message up to 75 characters per line.
And try reducing the commit title up to 50 characters.

chunkeey commented:
Thanks! Merged. 
(yes, I made the suggested changes to the commit message and included @bobafetthotmail suggestion as well.)
